WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersFlowSettingsViewController.swift

```diff
@@ -684,10 +695,16 @@ private extension BloggingRemindersFlowSettingsViewController {
     ///
     /// - Returns: A closure used to reset changes made to the prompt settings. Returns nil if the update condition is not fulfilled.
     func temporarilyUpdatePromptSettings() -> (() -> Void)? {
+        assert(Thread.isMainThread, "This function must be called from the main thread.")
+
+        // FIXME: Refactor this function to save asynchronously to avoid using the main context.
+        let context = coreDataStack.mainContext
```

Comment:
@crazytonyli is this something that should be done as a followup PR?

Reply:
I could. There are other people looking at the blog reminder area at the moment, but I think a refactor to this function alone shouldn't cause a clash with them.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/21016/commits/d2ac846f6660dbd5b1d7eae1e9d2cd87aea0e896